ci-operator/step-registry/telcov10n/functional/cnf-network/ocp-operators/telcov10n-functional-cnf-network-ocp-operators-commands.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+set -e
+set -o pipefail
+
+process_inventory() {
+    local directory="$1"
+    local dest_file="$2"
+
+    if [ -z "$directory" ]; then
+        echo "Usage: process_inventory <directory> <dest_file>"
+        return 1
+    fi
+
+    if [ ! -d "$directory" ]; then
+        echo "Error: '$directory' is not a valid directory"
+        return 1
+    fi
+
+    find "$directory" -type f | while IFS= read -r filename; do
+        if [[ $filename == *"secretsync-vault-source-path"* ]]; then
+          continue
+        else
+          echo "$(basename ${filename})": \'"$(cat $filename)"\'
+        fi
+    done > $dest_file
+
+    echo "Processing complete. Check ${dest_file}"
+}
+
+echo "Create group_vars directory"
+mkdir /eco-ci-cd/inventories/ocp-deployment/group_vars
+
+find /var/group_variables/common/ -mindepth 1 -type d | while read -r dir; do
+    echo "Process group inventory file: ${dir}"
+    process_inventory $dir /eco-ci-cd/inventories/ocp-deployment/group_vars/"$(basename ${dir})"
+done
+
+echo "Create host_vars directory"
+mkdir /eco-ci-cd/inventories/ocp-deployment/host_vars
+
+find /var/host_variables/${CLUSTER_NAME}/ -mindepth 1 -type d | while read -r dir; do
+    echo "Process host inventory file: ${dir}"
+    process_inventory $dir /eco-ci-cd/inventories/ocp-deployment/host_vars/"$(basename ${dir})"
+done
```

Comment:
Why do we need this part? I think we only need kubeconfig and bastion details to be able to install operators

Reply:
Yes you are right, we need only host bastion and group all details. That's why I mount only following secrets:

Reply:
Is there any way we can make it more generic? assuming we won't always use kni-qe-92 as our cluster name...

Reply:
There are 2 options:
1 - Use out inventory container
2 - Mount all clusters secrets and select the right one in bash script.

Reply:
Actually we can also add if/else condition to bash script.

if cluster_name == 'kni-qe-92' then;
echo "ansible_host: 1.1.1.1" > ./hosts_vars/bastion

In this case we'll need to extend this condition for each cluster.

Reply:
WDYT about saving the bastion inventory/address/key during the cluster deployment on `${SHARED_DIR}/bastion`?
This way, the only thing you need is kubeconfig, and maybe we can also save it there as well